Source/WebCore/page/IdentifierProvider.cpp

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2022 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "IdentifierProvider.h"
+
+//#include "RegistrableDomain.h"
+
+namespace WebCore {
+
+void IdentifierProvider::setCanvasIdentifier() {
+    // FIXME: Take canvas content, hash it, and combine it with domainIdentifier to create canvasIdentifier
+    WTFLogAlways("");
+}
+
+// FIXME: Domain hash must be salted.
+void IdentifierProvider::setDomainIdentifier(const unsigned hashedDomain) {
+    m_domainIdentifier = hashedDomain;
+}
```

Comment:
Function braces should be in their own line
```suggestion
void IdentifierProvider::setCanvasIdentifier()
{
    // FIXME: Take canvas content, hash it, and combine it with domainIdentifier to create canvasIdentifier
    WTFLogAlways("");
}

// FIXME: Domain hash must be salted.
void IdentifierProvider::setDomainIdentifier(const unsigned hashedDomain)
{
    m_domainIdentifier = hashedDomain;
}
```